src/lib/FlatGeobuf.svelte

```diff
@@ -0,0 +1,138 @@
+<script lang="ts">
+  import type { Snippet } from 'svelte';
+  import type { GeoJSONSource, SourceSpecification } from 'maplibre-gl';
+  import type { GeoJSON, Polygon, FeatureCollection } from 'geojson';
+  import type { HeaderMeta } from 'flatgeobuf';
+
+  import { onDestroy } from 'svelte';
+
+  import MapLibreGeoJSON from './GeoJSON.svelte';
+  import { addSource, removeSource } from './source';
+  import { getId, updatedSourceContext } from './context';
+  import { flatgeobufToGeoJson, filterGeomsCentroidsWithin } from './flatgeobuf';
+
+  type bboxType = [number, number, number, number];
+  interface Props {
+    id?: string;
+    url: string;
+    extent?: bboxType | Polygon | null;
+    extractGeomCols: boolean;
+    metadataFunc?: (headerMetadata: HeaderMeta) => void;
+    promoteId?: string;
+    maxzoom: number | undefined;
+    attribution: string | undefined;
+    buffer: number | undefined;
+    tolerance: number | undefined;
+    children?: Snippet;
+  }
+
+  let {
+    // Props for the flatgeobuf
+    id = getId('flatgeobuf'),
+    url,
+    extent,
+    extractGeomCols = false,
+    promoteId = undefined,
+    metadataFunc,
+    // Props to pass on to GeoJSON component
+    maxzoom,
+    attribution,
+    buffer,
+    tolerance,
+    children,
+  }: Props = $props();
+
+  const { map, self: sourceId } = updatedSourceContext();
+  let sourceObj: GeoJSONSource | undefined = $state();
+  let first = $state(true);
+  let geojsonData: GeoJSON = $state({ type: 'FeatureCollection', features: [] });
+
+  // Set currentSourceId as reactive property once determined from context
+  let currentSourceId: string | undefined = $state();
+  $effect(() => {
+    currentSourceId = $sourceId;
+  });
```

Comment:
This should be done with a $derived + $state combination. IIRC it looks like this

```
let currentSourceId = $derived(() => $state($sourceId))
```

That lets you update currentSourceId while ensuring that updates to `$sourceId` will overwrite it. It also might let you get rid of the `undefined` in the types.

I think Svelte 5 is eventually going to get better built-in support for this type of pattern.

Reply:
Hm actually this only seems to work with modifying properties inside an object, not for primitives. Never mind here.